File: alert_civique_back/src/main/java/com/enterprise/alert_civique/entity/LiveStream.java
package com.enterprise.alert_civique.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "lives_stream")
public class LiveStream {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "livestream_id")
    private Long livestreamId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private Users user;

    @Column(name = "started_at")
    private LocalDateTime startedAt;

    @Column(name = "ended_at")
    private LocalDateTime endedAt;

    @Column(name = "stream_url")
    private String streamUrl;

    @Column(name = "status")
    private String status;

    @Column(name = "video_url")
    private String videoUrl;

    @Column(name = "duration")
    private Integer duration;

    @PrePersist
    protected void onCreate() {
        startedAt = LocalDateTime.now();
        status = "LIVE";
    }
}

File: alert_civique_front/models/LiveStream.ts
// ─── Correspond à LiveStreamDTO (Java) ───────────────────────────────────────
// status Java : "LIVE" | "ENDED" | "SCHEDULED"
export interface LiveStream {
  livestreamId?: number;  // Long
  userId?: string;        // String (mapped "dummy-user" en attendant FK user)
  startedAt?: string;     // ISO LocalDateTime
  endedAt?: string;       // ISO LocalDateTime
  streamUrl?: string;
  status?: 'LIVE' | 'ENDED' | 'SCHEDULED';
  videoUrl?: string;      // URL de la vidéo uploadée
  mediaId?: number;       // Long FK vers Media
  duration?: number;      // Integer (secondes)
}
